x/stakeibc/keeper/unbonding_records_get_host_zone_unbondings_msgs_test.go

```diff
@@ -100,108 +109,349 @@ func (s *KeeperTestSuite) SetupGetHostZoneUnbondingMsgs() GetHostZoneUnbondingMs
+	name := "Wrong chain id"
+	var test = defaultUnbondingTestCase
+	test.chainId = "nonExistentChainId"
+	test.expectPass = false
+	s.VerifyTestCase_TestGetHostZoneUnbondingMsgs(name, test)
+}
+func (s *KeeperTestSuite) TestGetHostZoneUnbondingMsgs_NoEpochUnbondingRecords() {
+	name := "No epoch unbonding records"
+	var test = defaultUnbondingTestCase
+	test.epochUnbondingRecords = []recordtypes.EpochUnbondingRecord{}
+	test.expectPass = false
+	s.VerifyTestCase_TestGetHostZoneUnbondingMsgs(name, test)
+}
+func (s *KeeperTestSuite) TestGetHostZoneUnbondingMsgs_UnbodingTooMuch() {
+	name := "Unbonding too much"
+	var test = defaultUnbondingTestCase
+	test.validators = []*stakeibc.Validator{
+		{
+			Address:       hostVal1Addr,
+			DelegationAmt: 0,
+			Weight:        wgtVal1,
+		},
+		{
+			Address:       hostVal2Addr,
+			DelegationAmt: 0,
+			Weight:        wgtVal2,
+		},
+		{
+			Address: hostVal3Addr,
+			// DelegationAmt and Weight are the same as Val2, to test tie breaking
+			DelegationAmt: 0,
+			Weight:        wgtVal2,
+		},
+	}
+	test.expectPass = false
+	test.expectErr = sdkerrors.Wrap(sdkerrors.ErrNotFound, fmt.Sprintf("Could not unbond %d on Host Zone %s, unable to balance the unbond amount across validators", uint64(2_000_000), "GAIA"))
+	s.VerifyTestCase_TestGetHostZoneUnbondingMsgs(name, test)
+}
+func (s *KeeperTestSuite) TestGetHostZoneUnbondingMsgs_NoNonzeroWeightValidator() {
+	name := "No non-zero weight validator"
+	var test = defaultUnbondingTestCase
+	test.validators = []*stakeibc.Validator{
+		{
+			Address:       hostVal1Addr,
+			DelegationAmt: amtVal1,
+			Weight:        0,
+		},
+		{
+			Address:       hostVal2Addr,
+			DelegationAmt: amtVal2,
+			Weight:        0,
+		},
+		{
+			Address:       hostVal3Addr,
+			DelegationAmt: amtVal2,
+			Weight:        0,
+		},
+	}
+	test.expectErr = sdkerrors.Wrap(stakeibc.ErrNoValidatorAmts, fmt.Sprintf("Error getting target val amts for host zone %s %d: no non-zero validator weights", "GAIA", uint64(2_000_000)))
+	test.expectPass = false
 
-	// TODO: check epoch unbonding record ids here
-	actualUnbondMsgs, actualAmtToUnbond, actualCallbackArgs, _, err := s.App.StakeibcKeeper.GetHostZoneUnbondingMsgs(s.Ctx(), tc.hostZone)
-	s.Require().NoError(err)
+	s.VerifyTestCase_TestGetHostZoneUnbondingMsgs(name, test)
+}
 
-	// verify the callback attributes are as expected
-	actualCallbackResult, err := s.App.StakeibcKeeper.UnmarshalUndelegateCallbackArgs(s.Ctx(), actualCallbackArgs)
-	s.Require().NoError(err, "could unmarshal undelegation callback args")
-	s.Require().Equal(len(tc.hostZone.Validators), len(actualCallbackResult.SplitDelegations), "number of split delegations in success unbonding case")
-	s.Require().Equal(tc.hostZone.ChainId, actualCallbackResult.HostZoneId, "host zone id in success unbonding case")
+func (s *KeeperTestSuite) VerifyTestCase_TestGetUnbondingAmountAndRecords(name string, tc GetHostZoneUnbondingMsgsTestCase) {
```

Comment:
for `GetUnbondingAmountAndRecords` we want to test that we're only summing up records with status UNBONDING_QUEUE. So you can add to the default case some epoch unbonding records in various status's and then this should check that we are only summing some of them